build_tools/github_actions/post_build_upload.py

```diff
@@ -226,6 +230,15 @@ def upload_build_summary(args):
 
 
 def run(args):
+    if os.path.isdir(args.build_dir) == False:
```

Comment:
nit: `args.build_dir` is of type `pathlib.Path`, so you can use `Path.is_dir()` here
```suggestion
    if not args.build_dir.is_dir():
```

https://docs.python.org/3/library/pathlib.html#corresponding-tools